docs/query-metadata-style-guide.md

```diff
@@ -0,0 +1,173 @@
+
+*   `@id js/angular-js/missing-explicit-injection`
+*   `@id js/angular-js/duplicate-dependency`
+
+Note, `@id` properties should be consistent for queries that highlight the same issue for different languages. For example, the following queries identify format strings that contain unsanitized input in Java and C++ code respectively:
+
+
+
+*   `@id java/tainted-format-string`
+*   `@id cpp/tainted-format-string`
+
+
+### Query type `@kind`
+
+`@kind` is a required property that defines the type of query. The main query types are:
+
+
+
+*   alerts (`@kind problem`)
+*   alerts containing path information (`@kind path-problem`)
+
+These `@kind` properties support two further mandatory properties which are added by Semmle after the query has been tested, prior to deployment to LGTM. The following information is for reference:
+
+
+
+*   `@precision`–broadly indicates the proportion of query results that are true positives, while also considering their context and relevance:
+    *   `low `
+    *   `medium `
+    *   `high `
+    *   `very high`
+*   `@problem.severity`–defines the level of severity of the alert: 
+    *   `error`–an issue that is likely to cause incorrect program behavior, for example a crash or vulnerability.
+    *   `warning`–an issue that indicates a potential problem in the code, or makes the code fragile if another (unrelated) part of code is changed.
+    *   `recommendation`–an issue where the code behaves correctly, but it could be improved.
+
+The values of `@precision` and `@problem.severity` assigned to a query that is part of the standard set determine how the results are displayed by LGTM. See [About alerts](https://help.semmle.com/lgtm-enterprise/user/help/about-alerts.html) and [Alert interest](https://lgtm.com/help/lgtm/alert-interest) for further information. For information about using custom queries in LGTM on a 'per-project' basis, see [Writing custom queries to include in LGTM analysis](https://lgtm.com/help/lgtm/writing-custom-queries) and [About adding custom queries](https://help.semmle.com/lgtm-enterprise/admin/help/about-adding-custom-queries.html). 
+
+## Query tags `@tags`
+
+The `@tags` property is used to define categories that the query relates to. Each query should belong to one (or more, if necessary) of the following four top-level categories:
+
+*   `@tags correctness`–for queries that detect incorrect program behavior.
+*   `@tags maintainability`–for queries that detect patterns that make it harder for developers to make changes to the code.
+*   `@tags readability`–for queries that detect confusing patterns that make it harder for developers to read the code.
+*   `@tags security`–for queries that detect security weaknesses. See below for further information.
+
+There are also more specific `@tags` that can be specified. See, the following pages for more information on the low-level tags:
```

Comment:
Those pages don't give any information about the tags, they just show the range of tags that are currently defined (albeit with some interpretation by the Confluence uploader, like splitting on slashes). These URLs will need to change to link to help.semmle.com/wiki - the current links are to internal-only preview pages.

Reply:
True about the lack of information on these pages. I've reworded slightly to simply say 'see these pages for more examples'.